code/modules/atmospherics/machinery/airalarm.dm

```diff
@@ -816,7 +841,12 @@
 
 /obj/machinery/airalarm/power_change()
 	..()
+	if(stat & NOPOWER)
+		set_light(0)
+	else
+		set_light(brightness_on)
```

Comment:
update_icon() called just below also sets the light, you have no reason to do it here